Android/app/src/main/java/app/intra/net/go/GoVpnAdapter.java

```diff
@@ -112,20 +106,19 @@ private void connectTunnel() {
     final String fakeDns = FAKE_DNS_IP + ":" + DNS_DEFAULT_PORT;
 
     // Strip leading "/" from ip:port string.
-    String trueDns = resolver.getAddress().toString().substring(1);
     listener = new GoIntraListener(vpnService);
     String dohURL = PersistentState.getServerUrl(vpnService);
 
     try {
       LogWrapper.log(Log.INFO, LOG_TAG, "Starting go-tun2socks");
-      Transport transport = RemoteConfig.getUseGoDoh() ? makeDohTransport(dohURL) : null;
-      final IntraTunnel t = Tun2socks.connectIntraTunnel(tunFd.getFd(), fakeDns, trueDns, trueDns,
+      Transport transport = makeDohTransport(dohURL);
+      final IntraTunnel t = Tun2socks.connectIntraTunnel(tunFd.getFd(), fakeDns,
```

Comment:
nit: you could assign `tunnel` directly.

Reply:
Done